### PR TITLE
Remove cruft related to FIRESTORE_STUB_BUILD.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

**Comment by ehsannas** [comment]
It turns out that the CMake files don't have any reference to FIRESTORE_STUB_BUILD, so it's never been built when CMake was used.

File: firestore/src/common/futures.h [hunk 1]
```diff
@@ -47,23 +47,14 @@ Future<T> FailedFuture(Error error, const char* message) {
...
+ * Returns a failed future suitable for returning from an "invalid" instance.
  */
 template <typename T>
 Future<T> FailedFuture() {
   static auto* future = new Future<T>(FailedFuture<T>(
       Error::kErrorFailedPrecondition,
-      "This instance is in an invalid state. This could either because the "
-      "underlying Firestore instance has been destructed or because you're "
-      "running on an unsupported platform."));
+      "This instance is in an invalid state. This is because the  underlying "
```
Comment (var-const) [inline firestore/src/common/futures.h:56]: Ultranit: double space between `the` and `underlying`.

**Review by var-const — DISMISSED** [review]

File: firestore/src/common/futures.h (hunk 1 above)
Comment (ehsannas) [reply firestore/src/common/futures.h]: Done

**Comment by ehsannas** [comment]
Looks like my IDE had failed to save a few of the files. Thank you, grep.

**Comment by ehsannas** [comment]
The test failure is on `FirebaseRemoteConfigTest.TestFetchInterval` which is failing on all PRs at the moment.

**Review by var-const — DISMISSED** [review]

**Comment by ehsannas** [comment]
@var-const I had to rebase against HEAD in order to get https://github.com/firebase/firebase-cpp-sdk/pull/490 so that all tests pass. Otherwise I couldn't merge. And now that I have rebased, it requires you to approve it again...

**Review by var-const — APPROVED** [review]

**Pull request merged**